lib/roo_on_rails/routemaster/lifecycle_events.rb

```diff
@@ -0,0 +1,50 @@
+require 'active_support/concern'
+require 'new_relic/agent'
+require 'roo_on_rails/routemaster/publishers'
+
+module RooOnRails
+  module Routemaster
+    module LifecycleEvents
+      extend ActiveSupport::Concern
+
+      ACTIVE_RECORD_TO_ROUTEMASTER_EVENT_MAP = {
+        create: :created,
+        update: :updated,
+        destroy: :deleted,
+        noop: :noop
+      }.freeze
+      private_constant :ACTIVE_RECORD_TO_ROUTEMASTER_EVENT_MAP
+
+      def publish_lifecycle_event(event)
+        publishers = Routemaster::Publishers.for(self, routemaster_event_type(event))
+        publishers.each do |publisher|
+          begin
+            publisher.publish!
+          rescue => e
+            NewRelic::Agent.notice_error(e)
+          end
+        end
+      end
+
+      private
+
+      def routemaster_event_type(event)
+        ACTIVE_RECORD_TO_ROUTEMASTER_EVENT_MAP[event].tap do |type|
+          raise "invalid lifecycle event '#{event}'" unless type
+        end
+      end
+
+      module ClassMethods
+        def publish_lifecycle_events(*events)
+          events ||= %i(create update destroy)
```

Comment:
Additionally, `described_class._commit_callbacks` will give you a list of registed callbacks. You should use this in your test.